packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -138,12 +140,20 @@ enum _MediaQueryAspect {
 class MediaQueryData {
   /// Creates data for a media query with explicit values.
   ///
-  /// Consider using [MediaQueryData.fromView] to create data based on a
-  /// [dart:ui.FlutterView].
+  /// In a typical application, calling this constructor directly is rarely
+  /// needed. Consider using [MediaQueryData.fromView] to create data based on a
+  /// [dart:ui.FlutterView], or [MediaQueryData.copyWith] to create a new copy
+  /// of [MediaQueryData] with updated properties from a base [MediaQueryData].
   const MediaQueryData({
     this.size = Size.zero,
     this.devicePixelRatio = 1.0,
-    this.textScaleFactor = 1.0,
+    @Deprecated(
+      'Use textScaler instead. '
+      'Use of textScaleFactor was deprecated in preparation for the upcoming nonlinear text scaling support. '
+      'This feature was deprecated after v3.12.0-2.0.pre.',
+    )
+    double textScaleFactor = 1.0,
+    TextScaler textScaler = const _UnspecifiedTextScaler(),
```

Comment:
```suggestion
    TextScaler textScaler = _defaultMediaQueryTextScaler,
```

Reply:
also maybe renamed _defaultMediaQueryTextScaler to _kUnspecifiedTextScaler

Reply:
done.